### DISCUSSION TITLE
SyntaxError: Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#56374] [Help] [answered]
### Summary

This code here:

/api/indexes.js

```
import { TextLoader } from "langchain/document_loaders/fs/text";
import { CharacterTextSplitter } from "langchain/dist/text_splitter";
import { OpenAIEmbeddings } from "langchain/embeddings/openai";
import { FaissStore } from "langchain/vectorstores/faiss";

let loader;
let docs;
let splitter;
let documents;
let embeddings;
let vectorStore;

const indexes = async (req, res) => {
  loader = new TextLoader("../../utils/restaurant.txt");
  docs = await loader.load();
  splitter = new CharacterTextSplitter({
    chunkSize: 200,
    chunkOverlap: 50,
  });
  documents = await splitter.splitDocuments(docs);
  console.log(documents);

  embeddings = new OpenAIEmbeddings();
  vectorStore = await FaissStore.fromDocuments(documents, embeddings);
  await vectorStore.save("./");

  res.status(200).json({ ok: ok });
};

export default indexes;
```

Run:
`node pages/api/indexes.js`

Error:

```
(node:21728) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectd\pages\api\indexes.js:1
import { TextLoader } from "langchain/document_loaders/fs/text";
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

Why would I be getting that error above? 

What command can I use just to run a single file ?

**Comment by icyJoseph** [comment]
The root problem there is that Node.js by default understands only CJS synxtas, which doesn't have `import` or `export` keywords.

Another problem is that `page/api/indexes.js` does nothing really. You don't invoke any function or anything. 

What are you trying to accomplish?

If you want to run ES modules, your file might need to end with `mjs`, or in the package.json for the project, you can add a `"type": "module"` property.

**Comment by Elindo586** [comment]
I change the package.json to `"type": "module"`

^^^  Would this mess up any of the already existing files in Next.js using import?

The above code is to create and save a `docstore.json `and a `faiss.index` at the root level... which it isn't doing anything at the moment..

**Reply by icyJoseph** [reply, marked as answer]
It depends on what else you have. If you are unsure don't do it now, and learn ESM vs CJS later.

Why not write a separate script. Maybe even place those outside the src folder, in case you have one. I mean if they are not part of the server runtime, but a preparation step? And use .mjs on them.
